Dockerfile

```diff
@@ -4,26 +4,72 @@ ENV S2I_GIT_VERSION="" \
     S2I_GIT_MAJOR="" \
     S2I_GIT_MINOR=""
 
-
 WORKDIR /tmp/source-to-image
 COPY . .
 
-ENV GOARCH="amd64"
-
-USER root
-
 RUN make
 
 #
 # Runner Image
 #
 
-FROM registry.redhat.io/ubi8/ubi
+FROM registry.access.redhat.com/ubi8/ubi
+
+ENV GOARCH="amd64" \
+    _BUILDAH_STARTED_IN_USERNS="" \
+    BUILDAH_ISOLATION="chroot" \
+    S2I_HOME="/s2i" \
+    S2I_CONTAINER_MANAGER="buildah"
 
-ENV GOARCH="amd64"
+RUN useradd --uid 1000 --groups root --home-dir $S2I_HOME s2i && \
+    mkdir /src && \
+    chown s2i:s2i /src
+
+RUN echo -en "\
+[AppStream]\n\
+name=CentOS AppStream\n\
+mirrorlist=http://mirrorlist.centos.org/?release=8&arch=x86_64&repo=AppStream&infra=\n\
+# baseurl=http://mirror.centos.org/$contentdir/$releasever/AppStream/$basearch/os/\n\
+gpgcheck=1\n\
+enabled=1\n\
+gpgkey=https://www.centos.org/keys/RPM-GPG-KEY-CentOS-Official\n\
+" > /etc/yum.repos.d/appstream.repo && \
+    cat /etc/yum.repos.d/appstream.repo && \
+    yum -y reinstall shadow-utils && \
+    yum -y install buildah fuse-overlayfs fuse3 git && \
+    rm -rf /var/cache /var/log/dnf* /var/log/yum.*
+
+RUN sed -i \
```

Comment:
any sort of documentation you all referenced when you worked this up?  ... if so, let's add it as a reference in a comment that precedes this line

this entire set of bash feels very specific/detailed and perhaps fragile as underling ubi evolves

Reply:
That's a direct copy of [`buildah` Dockerfile](https://github.com/containers/buildah/blob/11ad04e83d5206ccd367eed65becb7d0908d3f95/contrib/buildahimage/stable/Dockerfile#L17).

Reply:
I end up copying buildah upstream, however, I'm also inclined to hold the whole configuration file, instead of editing during container build process. What do you think?

Reply:
Not sure I know what `hold the whole configuration file` would entail @otaviof .... is it possible to elaborate.

Otherwise, I would add the https://github.com/containers/buildah/blob/11ad04e83d5206ccd367eed65becb7d0908d3f95/contrib/buildahimage/stable/Dockerfile#L17 link you posted here as part of a comment in this file, explaining where it came from.

Reply:
Okay, I meant having the actual configuration file in this repository, and later `COPY` in the image. But sure, it indeed deserves a reference comment registering the origin.